backend/src/main/java/com/huellapositiva/domain/actions/ResendEmailConfirmationAction.java

```diff
@@ -38,7 +40,7 @@ public void execute() {
 
         boolean isEmailNotConfirmed = !credential.getEmailConfirmed();
         if (isEmailNotConfirmed) {
-            Integer updateOperation = jpaEmailConfirmationRepository.updateHashByEmail(email, Token.createToken().toString());
+            Integer updateOperation = jpaEmailConfirmationRepository.updateHashByEmail(email, Token.createToken().toString(), new Date());
```

Comment:
No need to add the date, hibernate should take care.
You need to create a new column in the EmailConfirmation entity called updateOn and annotate it with @UpdateTimestamp so that hibernate changes the date when the entity is updated automatically. Additionally you will need to declare that column in a new incremental flyway script

Reply:
It can also be fixed by changing the type of createdOn from _java.util.Date_ to _java.time.Instant_ and using the annotation **@PrePersist** to set its value before inserting the tuple on the DB and @UpdateTimestamp to let Hibernate update its value. This way, there is no need to add more columns in the table and/or create a flyway script (The Instant-Timestamp conversion is supported by default)

Reply:
We do want more columns. CreatedOn clearly declares an intention which is a ONE TIME date where an entity was created. For updates another column is used normally. Similarly, a person has only one date of birth. You cannot have two dates of birth the same way an entity cannot be created twice.